### PR TITLE
[coremedia] Add nullability to (generated and manual) bindings

### PR DESCRIPTION
**Pull request by tj-devel709**
This PR aims to bring nullability changes to CoreMedia.
Following the steps [here](https://github.com/xamarin/xamarin-macios/wiki/Nullability):

1. I am adding `nullable enable` to all manual files that are not "API_SOURCES" in src/frameworks.sources and making the required nullability changes

### CODE REVIEW

File: src/CoreMedia/CMCustomBlockAllocator.cs [hunk 1]
```diff
@@ -59,7 +62,9 @@ internal struct CMBlockBufferCustomBlockSource {
 		static IntPtr AllocateCallback (IntPtr refCon, nuint sizeInBytes)
 		{
 			var gch = GCHandle.FromIntPtr (refCon);
-			return ((CMCustomBlockAllocator) gch.Target).Allocate (sizeInBytes);
+			if (gch.Target is CMCustomBlockAllocator target)
+				return target.Allocate (sizeInBytes);
+			return IntPtr.Zero;
```
Comment (tj-devel709) [inline src/CoreMedia/CMCustomBlockAllocator.cs:67]: Not sure if this is correct

File: src/CoreMedia/CoreMedia.cs [hunk 2]
```diff
@@ -87,22 +90,24 @@ public struct CMTimeRange {
...
+#pragma warning restore CS0618 // Type or member is obsolete
 #endif
-			InvalidRange = (CMTimeRange)Marshal.PtrToStructure (retInvalid, typeof(CMTimeRange));
+			InvalidRange = (CMTimeRange)Marshal.PtrToStructure (retInvalid, typeof(CMTimeRange))!;
 
 			var retMappingInvalid = Dlfcn.dlsym (lib, "kCMTimeMappingInvalid");
-			if (retMappingInvalid  != IntPtr.Zero)
-				InvalidMapping = (CMTimeRange)Marshal.PtrToStructure (retMappingInvalid, typeof(CMTimeRange));
+			if (retMappingInvalid != IntPtr.Zero)
+				InvalidMapping = (CMTimeRange)Marshal.PtrToStructure (retMappingInvalid, typeof(CMTimeRange))!;
```
Comment (tj-devel709) [inline src/CoreMedia/CoreMedia.cs:110]: I was also not sure if these castings should be `!`. I tried making these CMTimeRange nullable, but I remember Rolf said not to make structs nullable

**Review by mandel-macaque — CHANGES_REQUESTED** [review]
This code does not compile:
```
Foundation/DictionaryContainer.cs(320,39): error CS8604: Possible null reference argument for parameter 'dict' in 'CMTime CMTime.FromDictionary(NSDictionary dict)'.

make[1]: *** [build/dotnet/maccatalyst/64/Microsoft.MacCatalyst.dll] Error 1

make[1]: *** Waiting for unfinished jobs....

Foundation/DictionaryContainer.cs(320,39): error CS8604: Possible null reference argument for parameter 'dict' in 'CMTime CMTime.FromDictionary(NSDictionary dict)'.

make[1]: *** [build/mac/mobile-64/Xamarin.Mac.dll] Error 1

Foundation/DictionaryContainer.cs(320,39): warning CS8604: Possible null reference argument for parameter 'dict' in 'CMTime CMTime.FromDictionary(NSDictionary dict)'.

Foundation/DictionaryContainer.cs(320,39): warning CS8604: Possible null reference argument for parameter 'dict' in 'CMTime CMTime.FromDictionary(NSDictionary dict)'.

Foundation/DictionaryContainer.cs(320,39): warning CS8604: Possible null reference argument for parameter 'dict' in 'CMTime CMTime.FromDictionary(NSDictionary dict)'.

Foundation/DictionaryContainer.cs(320,39): error CS8604: Possible null reference argument for parameter 'dict' in 'CMTime CMTime.FromDictionary(NSDictionary dict)'.
```

File: src/CoreMedia/CoreMedia.cs (hunk 2 above)
Comment (mandel-macaque) [reply src/CoreMedia/CoreMedia.cs:110]: For both of your questions, I would use the ! since we might be changing behaviour in both cases.

**Review by mandel-macaque — CHANGES_REQUESTED** [review]

File: src/CoreMedia/CMCustomBlockAllocator.cs (hunk 1 above)
Comment (rolfbjarne) [reply src/CoreMedia/CMCustomBlockAllocator.cs:67]: Looks good to me, can't be much worse than throwing a InvalidCastException.

**Review by rolfbjarne — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Comment by tj-devel709** [comment]
Unrelated Test Failure: https://github.com/xamarin/maccore/issues/2558

**Pull request merged**